Problem Statement 1:
Is gender independent of education level? A random sample of 395 people were
surveyed and each person was asked to report the highest education level they
obtained. The data that resulted from the survey is summarized in the following table:
High School Bachelors Masters Ph.d. Total
Female 60 54 46 41 201
Male 40 44 53 57 194
Total 100 98 99 98 395
Question: Are gender and education level dependent at 5% level of significance? In
other words, given the data collected above, is there a relationship between the
gender of an individual and the level of education that they have obtained?

In [49]:
import pandas as pd
f=[[60,50,46,41,201],[40,44,53,57,194],[100,98,99,98,395]]
title=['Highschool','bachleors','masters','phd','total']
 
df = pd.DataFrame(f, index=['Female','male','total'], columns =title) 
df

Highschool  bachleors  masters  phd  total
Female          60         50       46   41    201
male            40         44       53   57    194
total          100         98       99   98    395

In [50]:
df.shape

(3, 5)

In [54]:
lst=[]
i=0
for j in range(3):
    
    for k in range(5):
        #print(k)
        e=(df.get_value(2,k, takeable = True)*df.get_value(i,4, takeable = True))/df.get_value(2, 4, takeable = True)
        #print(i)
        i+=1
        if(i==3):
            i=0
        e=float("{:.2f}".format(e))
        lst.append(e)
    k=k+1
        

print(lst)
newl1=[lst[0],lst[3],lst[12],lst[6],lst[9]]
print(newl1)
newl2=[lst[10],lst[1],lst[7],lst[13],lst[4]]
print(newl2)


[50.89, 48.13, 99.0, 49.87, 194.0, 100.0, 49.87, 48.62, 98.0, 201.0, 49.11, 98.0, 50.38, 48.13, 395.0]
[50.89, 49.87, 50.38, 49.87, 201.0]
[49.11, 48.13, 48.62, 48.13, 194.0]


D:\anaconda\lib\site-packages\ipykernel_launcher.py:7: FutureWarning: get_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead
  import sys


In [55]:
f1=[newl1,newl2,[100,98,99,98,395]]
df = pd.DataFrame(f1, index=['Female','male','total'], columns =title) 
df

Highschool  bachleors  masters    phd  total
Female       50.89      49.87    50.38  49.87  201.0
male         49.11      48.13    48.62  48.13  194.0
total       100.00      98.00    99.00  98.00  395.0

In [56]:
f=[f[0],f[1]]
f1=[f1[0],f1[1]]

In [57]:
f

[[60, 50, 46, 41, 201], [40, 44, 53, 57, 194]]

In [58]:
import itertools

f = list(itertools.chain.from_iterable(f))
f1=list(itertools.chain.from_iterable(f1))


In [59]:
print(f,f1)

[60, 50, 46, 41, 201, 40, 44, 53, 57, 194] [50.89, 49.87, 50.38, 49.87, 201.0, 49.11, 48.13, 48.62, 48.13, 194.0]


In [60]:
x2=0
for i in range(len(f)):
    x2+=((float(f[i])-f1[i])**2)/f1[i]
    print(((float(f[i])-f1[i])**2)/f1[i])

1.6308135193554725
0.0003388810908361874
0.38079396585946845
1.5776398636454776
0.0
1.689922622683771
0.3543922709328905
0.3945783628136574
1.6346748389777677
0.0


In [61]:
x2

7.663154325359341

from the chisquare test table with confidence level 0.05 Degrees of freedom 3 the critical value is found to be 7.815

In [63]:
((50-49.87)**2)/49.87


0.0003388810908361874

since 7.66 < 7.88, I accept the null hypothesis and conclude that the education level independent on gender at a 5% level of significance.

but while doing manual calculation it seems the value comes around 8.006, In that case it should be rejected 

# Problem 2

In [183]:

g1=[51, 45, 33, 45, 67]
g2=[ 23, 43, 23, 43, 45]
g3=[ 56, 76, 74, 87, 56]
meang1=sum(g1)/5
meang2=sum(g2)/5
meang3=sum(g3)/5
meang3

69.8

In [184]:
devg2=list(map(lambda x:x-meang2, g2))
devg3=list(map(lambda x:x-meang3, g3))
devg1=list(map(lambda x:x-meang1, g1))

In [185]:
devg3

[-13.799999999999997,
 6.200000000000003,
 4.200000000000003,
 17.200000000000003,
 -13.799999999999997]

In [186]:
type(devg1)

list

In [215]:
sqdev1=list(map(lambda x:x**2,devg1))
sqdev2=list(map(lambda x:x**2,devg2))
sqdev3=list(map(lambda x:x**2,devg3))
sqdev1

[7.839999999999984,
 10.240000000000018,
 231.04000000000008,
 10.240000000000018,
 353.4399999999999]

In [258]:
ss1=sum(sqdev1)
ss2=sum(sqdev2)
ss3=sum(sqdev3)


515.2

In [261]:
ssmeans=(ss1+ss2+ss3)/3
ssmeans
print(ss1,ss2,ss3,ssmeans)

604.5866666666666 515.2 732.8 617.5288888888889


In [240]:
var1=ss1/(5-1) #no of data -1
var2=ss2/(5-1)
var3=ss3/(5-1)
var1

151.14666666666665

In [241]:
mse=(var1+var2+var3)/3

In [242]:
df=15-3 #15 values 3 group

In [243]:
sserror=mse/df

In [244]:
granmean=(meang1+meang2+meang3)/3 #mean of means

In [245]:
mean=[meang1, meang2,meang3]
grandmea=[granmean,granmean,granmean]

In [246]:
grandmea

[51.133333333333326, 51.133333333333326, 51.133333333333326]

In [247]:
dev=[devg1,devg2,devg3]
dev

[[2.799999999999997,
  -3.200000000000003,
  -15.200000000000003,
  -3.200000000000003,
  18.799999999999997],
 [-12.399999999999999,
  7.600000000000001,
  -12.399999999999999,
  7.600000000000001,
  9.600000000000001],
 [-13.799999999999997,
  6.200000000000003,
  4.200000000000003,
  17.200000000000003,
  -13.799999999999997]]

In [248]:
sqdev=[sqdev1,sqdev2,sqdev3]
sum(devg3)/5

2.842170943040401e-15

In [249]:
dev11=list(map(lambda x,y:x-y,mean,grandmea))

In [250]:
sqdev1=list(map(lambda x:x**2,dev11))


In [251]:
lst1=[mean,grandmea,dev11,sqdev1]

In [252]:
title=['mean','grandmean','deviation','square of Dev']
 
df1= pd.DataFrame(lst1) 
df1

0           1           2
0  48.200000   35.400000   69.800000
1  51.133333   51.133333   51.133333
2  -2.933333  -15.733333   18.666667
3   8.604444  247.537778  348.444444

In [253]:
df1=df1.transpose()

In [254]:
df1.columns=title

In [255]:
df1

mean  grandmean  deviation  square of Dev
0  48.2  51.133333  -2.933333       8.604444
1  35.4  51.133333 -15.733333     247.537778
2  69.8  51.133333  18.666667     348.444444

In [257]:
m

617.5288888888889

In [235]:
varmean=ssmeans/(3-1)
varmean

310.1333333333333

In [236]:
mean_square_within=varmean*5 #no of values in a group is 5

In [237]:
deg_free=3-1 #3 groups-1

In [238]:
ss_group=mean_square_within*deg_free
ss_group

3101.333333333333

In [263]:
f_test=mean_square_within/mse

In [264]:
f_test

10.044334408106861

f(critical)(dof-1,15-3)=(2,12) from F- table it is found to be 3.89 therefore, reject H0, accept H1

# problem 3

In [270]:
from statistics import stdev
l1=[10,20,30,40,50]
l2=[5,10,15, 20, 25]
va=[]
lst=[l1,l2]
n=len(lst)
for i in range(len(lst)):
    mean=sum(lst[i])/n
    sd=stdev(lst[i])
    var=sd**2
    va.append(var)


In [271]:
va

[250.0, 62.5]

In [272]:
f_test=va[0]/va[1]

In [273]:
f_test

4.0